istioctl/cmd/analyze.go

```diff
@@ -189,7 +192,7 @@ func Analyze() *cobra.Command {
 				if err != nil {
 					return err
 				}
-				sa.AddRunningKubeSource(k)
+				sa.AddRunningKubeSourceWithRevision(k, revision)
```

Comment:
How does this work? If specifying the revision, some required resources not having the revision labels will be skipped, like services, which will lead to wrong analysis results.

Reply:
AddRunningKubeSource just calls the latter function like so. I am not changing any existing behavior but allowing the end user to specify their own revision
```yaml
// AddRunningKubeSource adds a source based on a running k8s cluster to the current IstiodAnalyzer
// Also tries to get mesh config from the running cluster, if it can
func (sa *IstiodAnalyzer) AddRunningKubeSource(c kubelib.Client) {
	sa.AddRunningKubeSourceWithRevision(c, "default")
}

```